src/BuildingRegistry.Api.BackOffice/BuildingUnit/BuildingUnitController-Plan.cs

```diff
@@ -60,17 +60,14 @@ public partial class BuildingUnitController
             {
                 return Accepted();
             }
+            catch (AggregateNotFoundException)
+            {
+                throw new ApiException(ValidationErrorMessages.BuildingNotFound, StatusCodes.Status404NotFound);
```

Comment:
If a building is not found, it cannot be a 404, but a 400. So a validation exception.